nixos/modules/programs/less.nix

```diff
@@ -111,7 +112,7 @@ in
     environment.systemPackages = [ pkgs.less ];
 
     environment.variables = {
-      LESSKEY_SYSTEM = toString lessKey;
+      LESSKEYIN_SYSTEM = toString lessKey;
```

Comment:
Even better would be to wrap less to set that env var as a default for less and directly take effect on generation switches. It's not a requirement for this PR to be merged in.

Reply:
I'm a little hesitant to do that because then this change would affect non-NixOS users too, overriding any `LESSKEYIN_SYSTEM` they might have set in the host OS.

Reply:
Good call, wrapProgram has a [--set-default](https://github.com/jerith666/nixpkgs/blob/d4421b66754147e34e6167265d3548d3a74fa2a4/pkgs/build-support/setup-hooks/make-wrapper.sh#L18-L19)
option so that the env var can still be overwritten.